controllers/argocd/policyrule.go

```diff
@@ -209,6 +209,27 @@ func policyRuleForNotificationsController() []v1.PolicyRule {
 	}
 }
 
+func policyRuleForServerSupportedNamespaces() []v1.PolicyRule {
```

Comment:
@ishitasequeira 
if a namespace is already being managed by argo-cd (has the managed-by label) then that namespace will already have a server role with permissions to manage applications (https://github.com/argoproj-labs/argocd-operator/blob/master/controllers/argocd/policyrule.go#L117)
this might lead to redundancy if we create a separate role with a subset of the same permissions

do we want to first check if this is the case?